force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +236,13 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
+        if (d1 == null || d2 == null) {
+            return null;
+        }
+    
         // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        Integer num = d2.year() - d1.year();
```

Comment:
What if d2 is 1970 and d1 is 2024? How are you handling negative numbers (since it asks for the positive number of years between the two dates)? What Math method could help here?